internal/resolver/resolver.go

```diff
@@ -107,6 +110,27 @@ func (r *Resolver) Resolve(reqs []*chart.Dependency, repoNames map[string]string
 			continue
 		}
 
+		if strings.HasPrefix(d.Repository, "git://") {
```

Comment:
Perhaps all the open-coded `strings.HasPrefix(d.Repository, "git://")` calls could be moved into a function, e.g., `gitutils.IsGit`, analog to `registry.IsOCI`.